test/java/src/com/ibm/streamsx/rest/test/ParallelOperatorsTest.java

```diff
@@ -0,0 +1,107 @@
+import org.junit.Test;
+
+import com.ibm.streamsx.rest.Operator;
+import com.ibm.streamsx.topology.TStream;
+import com.ibm.streamsx.topology.Topology;
+import com.ibm.streamsx.topology.context.StreamsContext;
+import com.ibm.streamsx.topology.context.StreamsContextFactory;
+
+public class ParallelOperatorsTest {
+
+    StreamsConnectionTest helper;
+
+    @After
+    public void removeJob() throws Exception {
+        helper.removeJob();
+    }
+
+    @Before
+    public void setupJob() throws Exception {
+        if (null == helper) {
+            helper = new StreamsConnectionTest();
+        }
+        helper.setupInstance();
+        if (helper.jobId == null) {
+            // avoid clashes with sub-class tests
+            Topology topology = new Topology(getClass().getSimpleName(), 
+                    "JobForRESTApiTest");
+
+            TStream<Integer> source = topology.periodicSource(() -> (int) (Math.random() * 5000 + 1), 200, TimeUnit.MILLISECONDS);
+            source.invocationName("IntegerPeriodicMultiSource");
+            TStream<Integer> parallelSource = source.parallel(3);
+            TStream<Integer> sourceDouble = parallelSource.map(StreamsConnectionTest.doubleNumber());
+            sourceDouble.invocationName("Outer");
+            TStream<Integer> encore = sourceDouble.parallel(2);
+            TStream<Integer> encoreDouble = encore.map(StreamsConnectionTest.doubleNumber());
+            encoreDouble.invocationName("Inner");
+            TStream<Integer> joinDouble = encoreDouble.endParallel();
+            TStream<Integer> sourceDoubleAgain = joinDouble.endParallel().isolate().map(StreamsConnectionTest.doubleNumber());
+            sourceDoubleAgain.invocationName("ZIntegerTransformInteger");
+
+            if (helper.testType.equals("DISTRIBUTED")) {
+                helper.jobId = StreamsContextFactory.getStreamsContext(StreamsContext.Type.DISTRIBUTED).submit(topology).get()
+                        .toString();
+            } else if (helper.testType.equals("STREAMING_ANALYTICS_SERVICE")) {
+                helper.jobId = StreamsContextFactory.getStreamsContext(StreamsContext.Type.STREAMING_ANALYTICS_SERVICE)
+                        .submit(topology).get().toString();
+            } else {
+                fail("This test should be skipped");
+            }
+
+            helper.job = helper.instance.getJob(helper.jobId);
+            helper.job.waitForHealthy(60, TimeUnit.SECONDS);
+
+            assertEquals("healthy", helper.job.getHealth());
+        }
+        System.out.println("jobId: " + helper.jobId + " is setup.");
+    }
+
+    @Test
+    public void testParallelOperators() throws Exception {
+        List<Operator> operators = helper.job.getOperators();
+        Iterator<Operator> curr = operators.iterator();
+
+        assertTrue("Missing source operator", curr.hasNext());
+        Operator op = curr.next();
+        assertEquals("IntegerPeriodicMultiSource", op.getLogicalName());
```

Comment:
Hm, does the REST API guarantee that operators will be returned in a particular order?

Reply:
It seems to be the case, given that the test passes, but I'm wondering if that's something we should count on for other tests.

Reply:
That surprised me too when I first looked at the existing tests, but it does. From the [Operator](https://www.ibm.com/support/knowledgecenter/en/SSCRJU_4.2.0/com.ibm.streams.restapi.doc/doc/restapis-operator.html) docs:

> ***name***
> A string that identifies the operator name. By default, when you retrieve a collection of operator resources, they are sorted according to this property value.

Reply:
So _that's_ the reasoning behind `ZIntegerTransformInteger`.